tests/Benchmark/BenchmarkTest.h

```diff
@@ -0,0 +1,138 @@
+﻿//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#include <TestFramework.h>
+#include <type_traits>
+#include <chrono>
+#include <numeric>
+#include <utility>
+#include <memory>
+#include "BenchmarkTestPublisher.h"
+
+namespace testing {
+class BenchmarkTestBase {
+public:
+    inline void Run();
+
+    // Number of times to run each test, default to 2
+    size_t GetRunCount() const {
+        return 2;
+    }
+
+    // Pre-run and Post-run equivalent to SetUp and TearDown per test case
+    // Setup and Tear down that is performed once per all tests should be done in the constructor/destructor
+    void PreRun() {
+        // Default do nothing
+    }
+
+    void PostRun() {
+        // Default do nothing
+    }
+};
+
+namespace internal {
+
+// Helper method shared between BenchmarkTestRunner classes
+// This is called to actually run and benchmark the tests
+// Can't be shared internally because parameterized tests need a different inheritance than non
+template <typename T>
+static void __RunTest(T& test) {
+    using Microseconds = double;
```

Comment:
duration doesn't have operator*, / defined so I needed to convert out to do the calucations for stdDev, which had me saving double rather than Microseconds for my values which I didn't do because I didn't want to obscure that they represent microseconds